lib/wifi_user/use_case/check_user_is_sponsee.rb

```diff
@@ -0,0 +1,20 @@
+class WifiUser::UseCase::CheckUserIsSponsee
+  def initialize(allowlist_checker:)
+    @allowlist_checker = allowlist_checker
+  end
+
+  def execute(contact)
+    sponsee = DB[:userdetails].where(contact:)&.first
+
+    return false unless sponsee
+    return false unless sponsee[:sponsor]
+
+    sponsor_email = sponsee[:sponsor]
+
+    allowlist_checker.execute(sponsor_email)[:success]
```

Comment:
This is probably not necessary? If the user has been saved correctly, it should have passed the allowlist already right?

Reply:
Thats correct **allowlist_checker** is used when creating User
 
My thinking was the database might not always have correct data.

Reply:
> I think you forgot the SMS journey for sponsored users as well. Sponsored users should be able to send an sms and get their credentials back. (
> 
> https://github.com/alphagov/govwifi-user-signup-api/blob/7c45f2c3c60f91f1032b1de695f0a62e4d0df204/app.rb#L67
> 
> )

The sms journey looks the same for Sponsees and normal User e.g. Both use the same Templates.